packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,241 @@
+import assert from "node:assert";
+
+import { Temporal } from "@js-temporal/polyfill";
+
+import { Browser } from "../browser-compat-data/browser";
+import { Compat, defaultCompat } from "../browser-compat-data/compat";
+import { feature } from "../browser-compat-data/feature";
+import { Release } from "../browser-compat-data/release";
+import { browsers } from "./core-browser-set";
+import { isFuture, toDateString, toHighDate } from "./date-utils";
+import { support } from "./support";
+
+interface FeatureSelector {
+  compatKeys: [string, ...string[]];
+  checkAncestors: boolean;
+}
+
+type BaselineStatus = "low" | "high" | false;
+type BaselineDate = string | null;
+
+interface SupportStatus {
+  compatKey?: string;
+  baseline: BaselineStatus;
+  baseline_low_date: BaselineDate;
+  baseline_high_date: BaselineDate;
+  support: Map<Browser, Release | undefined>;
+  toJSON: () => string;
+}
+
+/**
+ * Given a set of compat keys, compute the aggregate Baseline support ("high",
+ * "low" or false, dates, and releases) for those keys.
+ */
+export function computeBaseline(
+  featureSelector: FeatureSelector,
+  compat: Compat = defaultCompat,
+): SupportStatus {
+  const { compatKeys } = featureSelector;
+  const keys = featureSelector.checkAncestors
+    ? compatKeys.flatMap((key) => withAncestors(key, compat))
+    : compatKeys;
+
+  const statuses = keys.map((key) => calculate(key, compat));
+  const support = collateSupport(statuses.map((status) => status.support));
+
+  const keystoneDate = findKeystoneDate(
+    statuses.flatMap((s) => [...s.support.values()]),
+  );
+  const { baseline, baseline_low_date, baseline_high_date } =
+    keystoneDateToStatus(keystoneDate);
```

Comment:
This is the biggest, most important change: I now follow the Baseline definition more literally. First we find the ostensible keystone date, then use that to determine if it's low, high, or false.

This is nice because it works the same way for the individual compat keys as it does for the final summation.